### ISSUE TITLE
No output torch 2.3.0, python 3.10.12

### ISSUE THREAD

**Issue by jturner116** [thuml/depyf#19]
Very excited to try out Depyf, hoping you can point me in right direction for the following

```python
import torch
from torch import _dynamo as torchdynamo
from efficient_kan import KAN
from torch.profiler import profile, record_function, ProfilerActivity
import depyf

torch.set_float32_matmul_precision("high")

model = KAN([28 * 28, 64, 10])
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model.to(device)
model = torch.compile(model)

# Warmup
for _ in range(10):
    model(torch.randn(64, 28 * 28).to(device))


def main():
    for _ in range(100):
        model(torch.randn(64, 28 * 28).to(device))


if __name__ == "__main__":
    with depyf.prepare_debug("./depyf_debug"):
        main()
```
Running the previous with TORCH_COMPILE_DEBUG gives the expected torchinductor output, but depyf does not write anything to the `depyf_debug` directory. 

I also get `UserWarning: You are trying to debug `torch.compile`. Please make sure the code runs multiple times to cover all the possible branches.`, but I believe I am running it several times

Testing it in [this](https://github.com/jturner116/efficient-kan/blob/master/examples/depyf_profile.py) repo if model will help. 

Thanks for your help :)

**Comment by youkaichao** [comment]
Hi, can you try to move the warmup code into `main`? I think your code is already compiled before you use `depyf.prepare_debug` , so it cannot capture anything for you.

e.g.

```python
def main():
	# Warmup
	for _ in range(10):
	    model(torch.randn(64, 28 * 28).to(device))

    for _ in range(100):
        model(torch.randn(64, 28 * 28).to(device))
```

**Comment by jturner116** [comment]
Ahh facepalm, thanks

**Comment by youkaichao** [comment]
@jturner116 does it help then?

**Comment by jturner116** [comment]
@youkaichao Yeah works great, thank you sir!
